### ISSUE TITLE
Suggestion: config structure

### ISSUE THREAD

**Issue by VimbaTheFish** [Ylinor/Storyteller#3]
Some tips which u can use for making config (and dialogue making process) more compact and simple.

1. U don't need dialogs array while u use unique dialogs with own id. Instead you can create key-value store for it, like u see in example.
2. U don't need to use commas, it works without them. At least not in arrays.
3. U don't need color string, while Sponge method to make colors exists. Use 
TextSerializers.FORMATTING_CODE.deserialize(rawInput)
to automaticly replace all & in strings into §. So users can make multicolored stuff.
4. While all your actions consist of 2 strings u can just combine them in one string and then parse it, using either space, or whatever u want. I made an example inside of paste.
5. Create comments inside example files so users can see, what stuff is optional, what every string do and   which strings can they remove without breaking their config.

Ofc that's on you which tips will u use, but I recommend u at least remove color string. And it will be cool to create separate module with dialogs and actions, which dialogs can call, but that's depends on your plugin structure, so u know better what to do.

My example how could it look: https://pastebin.com/EK2BsMwh

**Comment by obr-09** [comment]
Hey, we found your suggestions usefull, so we implemented most of them in the v0.1.2 release.  
1. Key-value stores are indeed easier to read, so I edited the parser behavior while keeping backward compatibility
2. I removed commas in the example for clarity, but we will probably generate a json anyway when we will make our generating tool
3. The use of formatting codes is easier and allow some fancy stuff, thanks for pointing that out
4. 👍
5. I didn't include comments in this release but might do in an other one
